packages/langium/src/workspace/ast-descriptions.ts

```diff
@@ -131,8 +130,8 @@ export class DefaultReferenceDescriptionProvider implements ReferenceDescription
 
     protected createDescription(refInfo: ReferenceInfo): ReferenceDescription | undefined {
         const targetNodeDescr = refInfo.reference.$nodeDescription;
-        const refCstNode = refInfo.reference.$refNode;
-        if (!targetNodeDescr || !refCstNode) {
+        const refSegment = refInfo.container.$segments?.properties[refInfo.property]?.[refInfo.index ?? 0];
```

Comment:
is the length of the refSegments array (last in the chain) always > 0 such that the fallback index 0 will succeed?

Reply:
Luckily, JS doesn't do bounds checking and simply returns `undefined` if the array is empty. Since we need to check for undefined anyway (see next line), this shouldn't lead to any issues.

Reply:
Right! `[][-1]` also works in JS 🙃